cmp/store/impl/StoreFilterFieldImplModel.js

```diff
@@ -15,103 +15,114 @@ import {
 export class StoreFilterFieldImplModel {
 
+    model;
+    bind;
     gridModel;
     store;
-    filterBuffer;
+
+    filterBuffer = 200;
     filterOptions;
     onFilterChange;
     includeFields;
     excludeFields;
-    model;
-    bind;
-
-    @observable value = '';
 
     filter = null;
-    applyFilterFn = null;
+    bufferedApplyFilter= null;
 
     constructor({
+        model,
+        bind,
         gridModel,
         store,
         filterBuffer = 200,
-        filterOptions,
         onFilterChange,
+        filterOptions,
         includeFields,
-        excludeFields,
-        model,
-        bind
+        excludeFields
     }) {
+        this.model = model;
+        this.bind = bind;
         this.gridModel = gridModel;
         this.store = store;
         this.filterBuffer = filterBuffer;
-        this.filterOptions = filterOptions;
         this.onFilterChange = onFilterChange;
+        this.filterOptions = filterOptions;
         this.includeFields = includeFields;
         this.excludeFields = excludeFields;
-        this.model = model;
-        this.bind = bind;
 
-        warnIf(includeFields && excludeFields,
-            "Cannot specify both 'includeFields' and 'excludeFields' props."
-        );
         warnIf(!gridModel && !store && isEmpty(includeFields),
-            "Must specify one of 'gridModel', 'store', or 'includeFields' or the filter will be a no-op"
+            "Must specify one of 'gridModel', 'store', or 'includeFields' or the filter will be a no-op."
         );
+        throwIf(!store && !bind, "Must specify either 'bind' or a 'store' in StoreFilterField.");
+
+        this.bufferedApplyFilter = debounce(() => this.applyFilter(), filterBuffer);
+
+        this.addReaction({
+            track: () => [this.filterText, gridModel?.columns, gridModel?.groupBy],
+            run: () => this.regenerateFilter(),
+            fireImmediately: true
+        });
+    }
+
+    // We allow these to be dynamic by updating on every render.
+    updateFilterProps({
+        onFilterChange,
+        filterOptions,
+        includeFields,
+        excludeFields
+    }) {
+        // just record change to callback
+        this.onFilterChange = onFilterChange;
 
-        if (store) {
-            this.applyFilterFn = debounce(
-                () => this.applyStoreFilter(),
-                filterBuffer
-            );
-
-            if (gridModel) {
-                this.addReaction({
-                    track: () => [gridModel.columns, gridModel.groupBy, filterOptions],
-                    run: () => this.regenerateFilter({applyImmediately: false})
-                });
-            }
+        // ...other changes require re-generation
+        if (!isEqual(
+            [filterOptions, includeFields, excludeFields],
+            [this.filterOptions, this.includeFields, this.excludeFields])
+        ) {
+            warnIf(includeFields && excludeFields, "Do not specify both 'includeFields' and 'excludeFields'");
```

Comment:
New warning here?  I _think_ technically this could be valid w/grid column visibility awareness (include foo field always even if not visible, exclude bar field always even if visible) or dot separated fields (`includeFields: ['user'], excludeFields: ['user.department']`)

If it is important to prevent should we have a setter so we could warn in the ctor as well?

Reply:
Not a new warning, but agree belonged in CTOR, and maybe obsolete. Fixed!